### PR TITLE
[ios] Adaptive fullscreen in landscape by device orientation

### PR DESCRIPTION
**Pull request by r0b0t3d**
Currently, when setting `fullscreenOrientation` to `landscape`, the video will always display in `landscape right`.
This PR allows checking the device's orientation then set the right `UIInterfaceOrientation`.

### CODE REVIEW

**Comment by sayem314** [comment]
Why is this ignored!

File: ios/Video/RCTVideoPlayerViewController.m [hunk 1]
```diff
@@ -28,7 +28,8 @@ - (UIInterfaceOrientationMask)supportedInterfaceOrientations {
 
 - (UIInterfaceOrientation)preferredInterfaceOrientationForPresentation {
   if ([self.preferredOrientation.lowercaseString isEqualToString:@"landscape"]) {
-    return UIInterfaceOrientationLandscapeRight;
+    UIDeviceOrientation currentOrientation = [[UIDevice currentDevice] orientation];
```
Comment (jensellfors) [inline ios/Video/RCTVideoPlayerViewController.m]: Not sure but I think you can just return `UIInterfaceOrientationLandscapeRight | UIInterfaceOrientationLandscapeLeft` no?

Comment (r0b0t3d) [reply ios/Video/RCTVideoPlayerViewController.m]: Yep, tested on simulator and working fine. I am going to update PR

**Comment by r0b0t3d** [comment]
@jenshandersson I got error in ios 12.x

Simulator Screen Shot - iPhone 5s - 2020-06-06 at 20 43 21

The error cause by this line 
```
return UIInterfaceOrientationLandscapeLeft | UIInterfaceOrientationLandscapeRight;
```
Fixed by changing to
```
UIDeviceOrientation currentOrientation = [[UIDevice currentDevice] orientation];
return currentOrientation == UIDeviceOrientationLandscapeRight ?
        UIInterfaceOrientationLandscapeLeft : UIInterfaceOrientationLandscapeRight;
```
Do you have any idea?

**Comment by jensellfors** [comment]
@r0b0t3d ah seems you can't return a mask for `preferredInterfaceOrientationForPresentation`... Let me push a fix for that, thanks

**Comment by jensellfors** [comment]
Hum this was tricker than I thought. I suggest we remove `fullscreenAutorotate` prop. And if you set `fullscreenOrientation="landscape"`we force landscape ie setting `supportedInterfaceOrientations` to `UIInterfaceOrientationMaskLandscape`. Thoughts @r0b0t3d?

**Comment by jensellfors** [comment]
@r0b0t3d I reverted this commit now. And using the snippet you sent won't work well when auto `fullscreenAutorotate` is true. There are a lot of permutations and edge cases we need to test...

**Comment by r0b0t3d** [comment]
@jenshandersson Agreed on point that we could remove `fullscreenAutorotate`.
So with `fullscreenOrientation`
- `all`: rotate by device orientations
- `landscape`: rotate to landscape left or right
- `portrait`

`preferredInterfaceOrientationForPresentation` return the orientation when presenting the view controller. So if we want auto rotate to landscape left or right, we must set `supportedInterfaceOrientations` to `UIInterfaceOrientationMaskLandscape` and `shouldAutorotate` to YES.

But this approach could works only if application support all the orientations. If application only support portrait, it gave this error `Supported orientations has no common orientation with the application, and [RCTVideoPlayerViewController shouldAutorotate] is returning YES`

I am doing another approach by implementation method swizzling `application:supportedInterfaceOrientationsForWindow:`. So we can override it with `UIInterfaceOrientationMaskAll` and do the logic on `supportedInterfaceOrientations`.

What do you think?

**Comment by jensellfors** [comment]
The more I'm thinking about it I feel we should just leave it to `all`... If you want lock or force an orientation, you can use https://github.com/wonday/react-native-orientation-locker (that's what I do). I feel this adds complexity ( and potential issues) to this project...

**Comment by r0b0t3d** [comment]
That sound is good. Doing my approach will make the project more complexity. Let me try with `react-native-orientation-locker`.
